src/libsync/syncfilestatustracker.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) by Klaas Freitag <freitag@owncloud.com>
+ * Copyright (C) by Jocelyn Turcotte <jturcotte@woboq.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#ifndef SYNCFILESTATUSTRACKER_H
+#define SYNCFILESTATUSTRACKER_H
+
+#include "ownsql.h"
+#include "syncfileitem.h"
+#include "syncfilestatus.h"
+#include <set>
+
+namespace OCC {
+
+class SyncEngine;
+
+struct Problem {
+    QString path;
+    SyncFileStatus::SyncFileStatusTag severity = SyncFileStatus::StatusError;
+
+    Problem(const QString &path) : path(path) { }
+    Problem(const QString &path, SyncFileStatus::SyncFileStatusTag severity) : path(path), severity(severity) { }
+
+    // Assume that each path will only have one severity, don't care about it in sorting
+    bool operator<(const Problem &other) const { return path < other.path; }
```

Comment:
it is better if the  operator< and operator== are non-member  (friends).  For simetry reason.

(eg,  you can type    "theProblem < myString"  but not  "myString < theProblem")